src/zc/ffscript.cpp

```diff
@@ -26344,49 +26304,7 @@ void set_register(int32_t arg, int32_t value)
 			} 
 			else 
 			{ 
-				switch(indx)
-				{
-					case 0: guysbuf[ri->npcdataref].misc1 = (value / 10000); break;
-					case 1: guysbuf[ri->npcdataref].misc2 = (value / 10000); break;
-					case 2: guysbuf[ri->npcdataref].misc3 = (value / 10000); break;
-					case 3: guysbuf[ri->npcdataref].misc4 = (value / 10000); break;
-					case 4: guysbuf[ri->npcdataref].misc5 = (value / 10000); break;
-					case 5: guysbuf[ri->npcdataref].misc6 = (value / 10000); break;
-					case 6: guysbuf[ri->npcdataref].misc7 = (value / 10000); break;
-					case 7: guysbuf[ri->npcdataref].misc8 = (value / 10000); break;
-					case 8: guysbuf[ri->npcdataref].misc9 = (value / 10000); break;
-					case 9: guysbuf[ri->npcdataref].misc10 = (value / 10000); break;
-					case 10: guysbuf[ri->npcdataref].misc11 = (value / 10000); break;
-					case 11: guysbuf[ri->npcdataref].misc12 = (value / 10000); break;
-					case 12: guysbuf[ri->npcdataref].misc13 = (value / 10000); break;
-					case 13: guysbuf[ri->npcdataref].misc14 = (value / 10000); break;
-					case 14: guysbuf[ri->npcdataref].misc15 = (value / 10000); break;
-					
-					case 15: guysbuf[ri->npcdataref].misc16 = value / 10000; break;
-					case 16: guysbuf[ri->npcdataref].misc17 = value / 10000; break;
-					case 17: guysbuf[ri->npcdataref].misc18 = value / 10000; break;
-					case 18: guysbuf[ri->npcdataref].misc19 = value / 10000; break;
-					case 19: guysbuf[ri->npcdataref].misc20 = value / 10000; break;
-					case 20: guysbuf[ri->npcdataref].misc21 = value / 10000; break;
-					case 21: guysbuf[ri->npcdataref].misc22 = value / 10000; break;
-					case 22: guysbuf[ri->npcdataref].misc23 = value / 10000; break;
-					case 23: guysbuf[ri->npcdataref].misc24 = value / 10000; break;
-					case 24: guysbuf[ri->npcdataref].misc25 = value / 10000; break;
-					case 25: guysbuf[ri->npcdataref].misc26 = value / 10000; break;
-					case 26: guysbuf[ri->npcdataref].misc27 = value / 10000; break;
-					case 27: guysbuf[ri->npcdataref].misc28 = value / 10000; break;
-					case 28: guysbuf[ri->npcdataref].misc29 = value / 10000; break;
-					case 29: guysbuf[ri->npcdataref].misc30 = value / 10000; break;
-					case 30: guysbuf[ri->npcdataref].misc31 = value / 10000; break;
-					case 31: guysbuf[ri->npcdataref].misc32 = value / 10000; break;
-					
-						default: 
-					{
-						Z_scripterrlog("Invalid Array Index passed to npcdata->Attributes[]: %d\n", (ri->npcdataref*10000)); 
-						break;
-					}
-				}
-					
+				guysbuf[ri->npcdataref].attributes[indx] = (value / 10000);					
```

Comment:
You removed the bounds checking
```suggestion
				if(indx < 0 || indx >= MAX_NPC_ATRIBUTES)
				{
					Z_scripterrlog("Invalid Array Index passed to npcdata->Attributes[]: %d\n", indx);
				}
				else guysbuf[ri->npcdataref].attributes[indx] = (value / 10000);					
```